### ISSUE TITLE
[Feature] Need Margin Ranking Loss

### ISSUE THREAD

**Issue by zx-sunnylife** [ml-explore/mlx#531]
I am currently working on implementing some recommendation system algorithms using MLX. During my work, I noticed that it does not include the [`MarginRankingLoss`](https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html) loss function like PyTorch, which is often critical for ranking tasks in recommendation systems.

Although I can implement this myself in my code, but maybe adding this loss would benefit other users who are working on recommendation systems and related areas?

**Comment by AtomicVar** [comment]
I'm willing to do this, if this is necessary. @awni

**Comment by awni** [comment]
Cool - I think this one is worth adding
